### PR TITLE
Creating a new set to avoid concurrent modification exception

### PR DESCRIPTION
**Pull request by jitsni**

### CODE REVIEW

File: src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/util/DelayedTaskScheduler.java [hunk 1]
```diff
@@ -52,11 +51,12 @@ public int process()
         {
             long c = System.currentTimeMillis();
             SortedSet<Long> past = scheduledTimes.headSet(c);
-            Iterator<Long> iter = past.iterator();
-            while(iter.hasNext())
+
+            past = new TreeSet<>(past);
+            scheduledTimes.removeAll(past);
+
+            for (Long time : past)
```
Comment (jfallows) [inline src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/util/DelayedTaskScheduler.java]: This is going to create a lot of garbage.

Comment (jitsni) [reply src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/util/DelayedTaskScheduler.java]: creating new TreeSet adds some garbage.

I think we have some cases where a task being executed adds a new task, hence concurrent modification exception. We need some collection which is not backed by the original collection (or add needs to defer adding to the scheduledTimes when there is an iterator is running. At the end of the iterator, we add all the deferred tasks)

Comment (jfallows) [reply src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/util/DelayedTaskScheduler.java]: Agreed. Recommend deferring the adds until after the iteration.

File: src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/util/DelayedTaskScheduler.java [hunk 2]
```diff
@@ -47,20 +69,34 @@ public void schedule(long time, Runnable task)
...
-
-            for (Long time : past)
+            Iterator<Long> iter = past.iterator();
+            while(iter.hasNext())
             {
+                Long time = iter.next();
+                iter.remove();
                 taskLookup.remove(time).run();
             }
             return past.size();
```
Comment (jfallows) [inline src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/util/DelayedTaskScheduler.java]: This `return` will prevent `schedulingDeferred` from being assigned back to `false` and deferred tasks from being added until next call to `process`.

**Pull request merged**